src/js/components/FormField/FormField.js

```diff
@@ -451,7 +458,21 @@ const FormField = forwardRef(
         {...outerProps}
         style={outerStyle}
         onFocus={(event) => {
-          setFocus(shouldKeepFocus());
+          if (!isDateInputComponent) {
+            setFocus(containsFocus(formFieldRef.current));
+          } else {
+            // if FormField contains a DateInput component check if the calendar
+            // button is in focus before assigning focus to the FormField
+            const formDescendants =
+              formFieldRef.current.getElementsByTagName('*');
+            let buttonChild;
+            for (let i = 0; i < formDescendants.length; i += 1) {
+              const child = formDescendants[i];
+              if (child.tagName.toLowerCase() === 'button') buttonChild = child;
```

Comment:
This looks easy to break if we change what types of components/html make up a DateInput